spec/rack_session_redis/store_spec.rb

```diff
@@ -0,0 +1,154 @@
+# frozen_string_literal: true
+
+RSpec.describe RackSessionRedis::Store do
+  let(:redis) { MockRedis.new }
+  let(:store) { described_class.new(redis: redis, prefix: 'rack_session') }
+
+  describe '.create' do
+    subject { described_class.create(url) }
+
+    let(:store) { instance_double(described_class) }
+
+    context 'when URL does not contain namespace' do
+      let(:url) { 'redis://localhost:6379/0' }
+
+      before do
+        expect(Redis).to receive(:new).with(url: URI(url)).and_return(redis)
+        expect(described_class).to receive(:new)
+          .with(redis: redis, prefix: 'sessions').and_return(store)
+      end
+
+      it { is_expected.to eq(store) }
+    end
+
+    context 'when URL contains namespace' do
+      let(:url) { 'redis://localhost:6379/0/rack_session' }
+
+      before do
+        expect(Redis).to receive(:new)
+          .with(url: URI('redis://localhost:6379/0')).and_return(redis)
+        expect(described_class).to receive(:new)
+          .with(redis: redis, prefix: 'rack_session').and_return(store)
+      end
+
+      it { is_expected.to eq(store) }
+    end
+  end
+
+  describe '#info' do
+    subject { store.info }
+
+    it { is_expected.to be_a(Hash) }
+  end
+
+  describe '#set' do
+    let(:data) { Marshal.dump({ a: 1 }) }
+
+    context 'when options is not provided' do
+      subject { store.set('key', { a: 1 }) }
+
+      it 'sets the value correctly' do
+        expect(redis).to receive(:set).with('rack_session:key', data, ex: described_class::DEFAULT_EXPIRES_IN)
+                                      .and_call_original
+
+        expect do
+          subject
+        end.to change { redis.get('rack_session:key') }.from(nil).to(data)
+      end
+    end
+
+    context 'when options is provided' do
+      subject { store.set('key', { a: 1 }, { expires_in: expires_in }) }
+
+      let(:expires_in) { 3600 }
+
+      it 'sets the value correctly' do
+        expect(redis).to receive(:set).with('rack_session:key', data, ex: expires_in)
+                                      .and_call_original
+
+        expect do
+          subject
+        end.to change { redis.get('rack_session:key') }.from(nil).to(data)
```

Comment:
Ah yes, we should.

Reply:
Actually, with the `expect` in line 66
```ruby
expect(redis).to receive(:set).with('rack_session:key', data, ex: expires_in).and_call_original
```
We should be sure that TTL is set correctly already, unless we worry that interface of redis-rb will change.

Reply:
That's true. I guess unless it was something like the `mset_ttl` where we introduced something new. Ah wells, I suppose it's good to have this pattern here so in case we extend with new methods in future we can follow this pattern to have assurance.